common/parallel/FileSystemCrawler.go

```diff
@@ -125,6 +129,7 @@ func enumerateOneFileSystemDirectory(dir Directory, enqueueDir func(Directory),
 
 	d, err := os.Open(dirString) // for directories, we don't need a special open with FILE_FLAG_BACKUP_SEMANTICS, because directory opening uses FindFirst which doesn't need that flag. https://blog.differentpla.net/blog/2007/05/25/findfirstfile-and-se_backup_name
 	if err != nil {
+		enqueueOutput(FileSystemEntry{dirString, nil}, err)
```

Comment:
nit, add a comment above this (and all such places)

// nil fileInfo value should mean that the FileSystemEntry refers to a directory.